youtube_dl/downloader/httpfd.py

```diff
@@ -0,0 +1,32 @@
+from __future__ import unicode_literals
+
+from .http import HttpFD
+
+try:
+    import urllib.request as compat_urllib_request
+except ImportError:  # Python 2
+    import urllib2 as compat_urllib_request
+import threading
+
+
+class HttpHB(HttpFD):
```

Comment:
The class name should ends with FD

Reply:
Thining again and I believe there's no need for a separate class. The heart beat function can be implemented in the FileDownloader class. (not in HttpFD as other stream types, including HLS and HDS, may also need heart beats)